src/lib/reasoners/enum_rel.ml

```diff
@@ -44,11 +46,26 @@ module HSS = Set.Make (struct type t=Hs.t let compare = Hs.compare end)
 
 module LR = Uf.LX
 
-type t = { mx : (HSS.t * Ex.t) MX.t; classes : Expr.Set.t list;
-           size_splits : Numbers.Q.t }
+type t = {
+  mx : (HSS.t * Ex.t) MX.t;
+  (* Map of uninterpreted enum semantic values to domains of their possible
+     values. The explanation justifies that the value of the semantic value
+     lies in the domain. *)
```

Comment:
(Looks like it is arbitrary semantic values)

Reply:
I'm not sure to understand what you mean by `normal form` here?  There is no canonization for Enum theory.

Reply:
I mean normal form modulo congruence (see `find_r` → `find_or_normal_form` in `Uf`), e.g. if we have `f(x)` and `f(x) = y` then the normal form of `f(x)` is `y`; in other word, class representatives (modulo AC(X) shenaningans).

Reply:
I am not asking to change the implementation, just trying to understand what it does so that the documentation is accurate :)

(Note that using class representatives as the key would also cause the same issue we observe in #474 / #823 )

Reply:
I am not saying to change the implementation in this PR ;)